Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -86,8 +86,25 @@ public object Create(IonSerializationOptions options, IIonReader reader, Type ta
             if (annotations.Length > 0)
             {
                 var typeName = annotations[0];
-                var assemblyName = options.AnnotatedTypeAssemblies.First(a => Type.GetType(FullName(typeName, a)) != null);
-                return Activator.CreateInstance(Type.GetType(FullName(typeName, assemblyName)));
+                Type typeToCreate = null;
+                try
+                {
+                    // First throws InvalidOperationException if no elements match the condition
+                    var assemblyName = options.AnnotatedTypeAssemblies.First(a => Type.GetType(FullName(typeName, a)) != null);
+                    typeToCreate = Type.GetType(FullName(typeName, assemblyName));
+                }
+                catch (InvalidOperationException)

```

Comment:
Processing by exception-handling is generally a bad idea. In this case it's very bad because this exception is so generic. Please refactor to use if-statements instead.

Reply:
Modified the logic to not rely on exception handling.